kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -238,12 +300,17 @@ namespace Kratos
 		// Initialize cut edges vectors and points arrays
 		unsigned int n_cut_edges = 0;
 		rIntersectionPointsArray.clear();
-		rCutEdgesVector = array_1d<unsigned int, n_edges>(n_edges, 0);
+		array_1d<unsigned int, n_edges> cut_edges_vector = array_1d<unsigned int, n_edges>(n_edges, 0);
```

Comment:
This is way inefficient. Instead of using the above reference to the TLS container array you are allocating an `array_1d` within the loop. The old behavior should be emulated.

Reply:
Sorry, I'll add the cut_edges_vector again.

Reply:
But I don't quite understand the reasoning. Isn't it in the loop anyway? The only difference I see is, that it is created later.

Reply:
That's precisely the point, by creating it later (within the loop) you're allocating the memory inside the loop. If creating it outside the memory is allocated outside the loop and inside the loop we only update the value. Considering that the expensive thing in creating is allocating the memory, it is much more efficient to do it outside the loop.

Reply:
Before I moved it, it was outside the function, but still inside the loop .. ?

Reply:
All of the variables the method gets are allocated inside the loop. Looping through the elements I mean, is there something else I am not aware of?

Reply:
Mmmm... I think I don't get your point. The idea is that (array) variables are declared (allocated) outside the loop and then used inside.

Reply:
Which loop do you mean? Are we talking about different loops? The only loop, that I am aware of and isn't just local, is the one that goes through all the elements of the ModelPart

What I try to say is:
-  `ComputeEdgesIntersections` gets `rCutEdgesRatioVector` and `rCutExtraEdgesRatioVector` and `rIntersectionPointsArray`, and previously `rCutEdgesVector` aswell, to "fill" with meaningful data
- all of these variables are allocated either in `CalculateElementalDistances` or `CalculateElementalAndEdgeDistances`, which then calls `ComputeEdgesIntersections` and both of these methods are already inside the loop (looping trough all elements in the ModelPart)
- So, the only thing I changed, that I am aware of at least, is to move the allocation from `CalculateElemental...Distances` to `ComputeEdgesIntersections`, because `rCutEdgesVector` is not used in `CalculateElemental...Distances`

Reply:
(discussed)